crates/polars-plan/src/plans/optimizer/predicate_pushdown/join.rs

```diff
@@ -149,7 +149,8 @@ pub(super) fn process_join(
 
     for (_, predicate) in acc_predicates {
         // Cross joins produce a cartesian product, so if a predicate combines columns from both tables, we should not push down.
-        if matches!(options.args.how, JoinType::Cross)
+        // Inequality joins logically produce a cartesian product, so the same logic applies.
+        if (matches!(options.args.how, JoinType::Cross) || options.args.how.is_ie())
```

Comment:
Let's add an `is_cross` as well. Looks a bit cleaner. :)

Reply:
Done. I used it in the other places we previously did the `matches!` dance as well.